### PR TITLE
Add deno_websocket

### PR DESCRIPTION
**Pull request by kovaszab**
This PR introduces WebSocket support by integrating the deno_websocket package.

### CODE REVIEW

**Comment by rscarson** [comment]
Looks great!

My only question would be weather we could feed it the options in web, since they are the same ones

**Comment by kovaszab** [comment]
I guess it makes more sense for them to use the same config. Now `WebSocketOptions` struct kinda looses sense though, maybe I should remove that too and only use `WebOptions`

**Comment by rscarson** [comment]
> I guess it makes more sense for them to use the same config. Now `WebSocketOptions` struct kinda looses sense though, maybe I should remove that too and only use `WebOptions`

Good idea, in practice having split configs might cause issues 

Pretty sure deno itself uses the same options for both

**Comment by rscarson** [comment]
Can you add the new example to the examples workflow in .gitHub?

After that looks great to me!
I appreciate the contribution, adding the extensions is very time consuming so I appreciate any help

**Comment by rscarson** [comment]
LGTM, thanks!

**Comment by kovaszab** [comment]
I like to help and also appreciate the work you put into implementing this. It's perfect for my use-case. I wrote the implementation of the extension because I needed a way to bridge JS and rust in an event-driven application. So the easiest solution was to WebSockets since I already run a server for it locally.

Off-topic, but I also experimented with implementing the `BroadcastChannel` extension, works great with one runtime with multiple handlers, however the deno implementation `InMemoryBroadcastChannel` can't be used with 2 runtimes at the sime time. I think this could be solved by creating a custom `BroadcastChannel` trait implementation utilizing `crossbeam-channel` crate.

So in the future I'll be working on implementing the `BroadcastChannels` properly and after that maybe I could make them be shared across runtimes optionally. I'll create the PR once I'm done with my experiments.

File: Cargo.toml [hunk 1]
```diff
@@ -59,6 +59,9 @@ web_stub = ["webidl"]
     # [https://html.spec.whatwg.org/multipage/webstorage.html]
     webstorage = ["deno_webstorage", "webidl"]
 
+    # [https://websockets.spec.whatwg.org/]
+    websocket = ["deno_websocket", "deno_web"]
```
Comment (rscarson) [inline Cargo.toml:63]: Oh wait hang on
This needs to depend on web not deno_web

```suggestion
    websocket = ["deno_websocket", "web"]
```

The web feature also lines up deno_web's dependencies

**Review by rscarson — CHANGES_REQUESTED** [review]

**Pull request merged**